### DISCUSSION TITLE
Is there a way to make one key's all properties notNull with a single annotation?

### DISCUSSION THREAD

**Discussion by EdwardJunyan** [victools/jsonschema-generator#378] [Q&A]
Here is the situation: 

```JAVA
class Position{
  private Double longitude;
  private Double latitude;
}

class Spot{
  private Position spotA;
  private Position spotB;
}
```

I only want to make `spotB` and spotB's `longitude` & `latitude` be `@NonNull`.

How to make it happen without creating unnecessary `Position` class?
Is there a way I put an annotation on the `spotB` and it's all properties will be nonNull?

Here is my goal
```JSON
{
  "properties": {
    "spotA": {
      "type": "object",
      "properties": {
        "longitude": {
          "type": "number"
        },
        "latitude": {
          "type": "number"
        }
      }
    },
    "spotB": {
      "type": "object",
      "properties": {
        "longitude": {
          "type": "number"
        },
        "latitude": {
          "type": "number"
        }
      },
      "required": ["longitude", "latitude"]
    }
  },
  "required": ["spotB"]
}
```

**Comment by CarstenWickner** [comment]
Hi @EdwardJunyan,

I've played around with this idea but couldn't come up with an easy solution from the top of my head.
For the given example, I'd question the meaningfulness of the endeavor (the `longitude` and `latitude` should always be mandatory, and only `spotA` is allowed to be omitted), but I can appreciate that this is only a minimal example of your actual challenge.

One idea would be to write a `CustomPropertyDefinitionProvider` that kicks in when `spotB` is being looked at and some kind of custom `@AllPropertiesMandatory` annotation is present on it.
However, one would have to make a couple of annotations around the kind of types this annotation is used, to not overcomplicate the implementation. If I wanted to provide a generic solution. That got me down the rabbit hole for a while, unfortunately.

Creating a separate subtype where all properties are mandatory seems the easier approach to me at this point. 🤔
